src/Illuminate/Support/Collection.php

```diff
@@ -844,6 +844,40 @@ public function implode($value, $glue = null)
         return implode($value, $this->items);
     }
 
+
+    /**
+     * splitting values by a string delimiter.
+     *
+     * @param  string  $value
+     * @param  string  $delimiter
+     * @return array
+     */
+    public function explode($delimiter = null, $value = null)
+    {
+        $first = $this->first();
+
+        if(is_null($delimiter))
+        {
+            return $this->toArray();
+        }
+
+        if(is_array($first) || is_object($first))
+        {
+            if(! is_null($value)) {
+                return $this->pluck($value)->map(function($item) use($delimiter) {
```

Comment:
is a pluck here needed? you could just map() the collection and get the item's key